x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/YamlTemplateRegistry.java

```diff
@@ -226,6 +247,17 @@ private IngestPipelineConfig loadIngestPipeline(String name, int version, @Nulla
         );
     }
 
+    // IndexTemplateRegistry ensures that ILM lifecycle policies are not loaded
+    // when in DSL only mode.
+    private LifecyclePolicy loadLifecyclePolicy(String name) {
+        try {
+            var rawPolicy = loadResource(this.getClass(), "/lifecycle-policies/" + name + ".yaml");
+            return LifecyclePolicyUtils.parsePolicy(rawPolicy, name, LifecyclePolicyConfig.DEFAULT_X_CONTENT_REGISTRY, XContentType.YAML);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
We usually use `ElasticsearchException` over RuntimeException, it should be a one-to-one swap here.

Reply:
Thanks, I have updated this (also updated a few other places where `RuntimeException` was being used in the file)